### PR TITLE
fix: Remove/fix API marked as removal for 2023-1

### LINKED ISSUE #394: Remove/fix API marked as removal for 2023-1
[kind/Enhancement]

## Which functionality do you think we should update/improve?
pluginVerifier indicates that there is at least 2 API that should be fixed: 
```
Deprecated API usages (1): 
    #Deprecated method com.intellij.ui.PopupHandler.installPopupHandler(JComponent, String, String) invocation
        Deprecated method com.intellij.ui.PopupHandler.installPopupHandler(javax.swing.JComponent component, java.lang.String groupId, java.lang.String place) : void is invoked in org.jboss.tools.intellij.openshift.WindowToolFactory.createToolWindowContent(Project, ToolWindow) : void. This method will be removed in 2022.1
Internal API usages (1): 
    #Internal method com.intellij.openapi.module.Module.getModuleFilePath() invocation
        Internal method com.intellij.openapi.module.Module.getModuleFilePath() : java.lang.String is invoked in org.jboss.tools.intellij.openshift.utils.ProjectUtils.getModuleRoot(Module) : VirtualFile. This method is marked with @org.jetbrains.annotations.ApiStatus.Internal annotation and indicates that the method is not supposed to be used in client code.
```

## Why is this needed? 
the first warning indicates that the plugin is not going to work with 2022-1

### LINKED ISSUE #515: Delete component in dev/debug mode does not clean cluster
[kind/bug]

## What versions of software are you using?
- Operating System:
- IntelliJ version:
- OpenShift version:

## What did you run exactly?
see https://issues.redhat.com/browse/JBIDE-28822
when deleting component in dev/debug mode, it's process isn't killed.

### LINKED ISSUE #519: Bug - exception on startup - failed to fetch api groups from kubernetes: Unauthorized"
[kind/bug]

## What versions of software are you using?
- Operating System:Windows
- IntelliJ version:
- OpenShift version:1.2.0.168

## What did you run exactly?
I started Intellij. I previously had connected it to the cluster and added components, it seems that it's giving issues with it now. Doesn't seem to gracefully handle these connection changes

## Any logs, error output, etc?
java.io.IOException: Process exited with an error: 1 (Exit value: 1) {
	"message": "unable to list all dynamic resources required to find components: failed to fetch api groups from kubernetes: Unauthorized"
}

	at com.redhat.devtools.intellij.common.utils.ExecHelper.executeWithResult(ExecHelper.java:261)
	at org.jboss.tools.intellij.openshift.utils.odo.OdoCli.execute(OdoCli.java:244)
	at org.jboss.tools.intellij.openshift.utils.odo.OdoCli.discover(OdoCli.java:856)
	at org.jboss.tools.intellij.openshift.utils.odo.OdoProjectDecorator.discover(OdoProjectDecorator.java:263)
	at org.jboss.tools.intellij.openshift.tree.application.ApplicationsRootNode.addContext(ApplicationsRootNode.java:138)
	at org.jboss.tools.intellij.openshift.tree.application.ApplicationsRootNode.loadProjectModel(ApplicationsRootNode.java:109)
	at org.jboss.tools.intellij.openshift.tree.application.ApplicationsRootNode.lambda$initializeOdo$0(ApplicationsRootNode.java:83)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2325)
	at org.jboss.tools.intellij.openshift.tree.application.ApplicationsRootNode.initializeOdo(ApplicationsRootNode.java:78)
	at org.jboss.tools.intellij.openshift.tree.application.ApplicationsTreeStructure.getApplicationsRoot(ApplicationsTreeStructure.java:87)
	at org.jboss.tools.intellij.openshift.tree.application.ApplicationsTreeStructure.getChildElements(ApplicationsTreeStructure.java:99)
	at com.intellij.ui.tree.StructureTreeModel.getValidChildren(StructureTreeModel.java:412)
	at com.intellij.ui.tree.StructureTreeModel.validateChildren(StructureTreeModel.java:330)
	at com.intellij.ui.tree.StructureTreeModel.getNode(StructureTreeModel.java:324)
	at com.intellij.ui.tree.StructureTreeModel.getChildren(StructureTreeModel.java:344)
	at com.intellij.ui.tree.AsyncTreeModel$CmdGetChildren.getNode(AsyncTreeModel.java:548)
	at com.intellij.ui.tree.AsyncTreeModel$Command.get(AsyncTreeModel.java:443)
	at com.intellij.ui.tree.AsyncTreeModel$Command.get(AsyncTreeModel.java:409)
	at com.intellij.util.concurrency.Invoker$Task.run(Invoker.java:305)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1111)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:71)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:63)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:128)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:109)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:68)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:68)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:106)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:71)
	at com.intellij.util.concurrency.Invoker.invokeSafely(Invoker.java:187)
	at com.intellij.util.concurrency.Invoker.lambda$offerSafely$0(Invoker.java:165)
	at com.intellij.util.concurrency.Invoker$Background.lambda$offer$0(Invoker.java:475)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:243)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:216)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:212)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:205)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)
	at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
	at com.redhat.devtools.intellij.common.utils.ExecHelper.executeWithResult(ExecHelper.java:258)
	... 51 more

### PR DESCRIPTION
**Pull request by sbouchet**
## What is the purpose of this change? What does it change?
Remove/fix API marked as removal for 2023-1
cleanup code
minimal supported version bump to 2021.2

## Was the change discussed in an issue?
fixes #394 
fixes #515 
fixes #519

### CODE REVIEW

File: src/main/java/org/jboss/tools/intellij/openshift/WindowToolFactory.java [hunk 1]
```diff
@@ -46,7 +47,7 @@ public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindo
         tree.putClientProperty(Constants.STRUCTURE_PROPERTY, structure);
         tree.setCellRenderer(new NodeRenderer());
         tree.setRootVisible(false);
-        PopupHandler.installPopupHandler(tree, "org.jboss.tools.intellij.tree", ActionPlaces.UNKNOWN);
+        PopupHandler.installPopupMenu(tree, "org.jboss.tools.intellij.tree", ActionPlaces.MAIN_MENU);
```
Comment (adietish) [inline src/main/java/org/jboss/tools/intellij/openshift/WindowToolFactory.java:50]: @sbouchet this fails to compile currently:
The method is only available in 212.4746.92 while our build here targets 212+
We have a compatibility wrapper in intellij-common to solve this (that I am using in intellij-kubernetes):
https://github.com/redhat-developer/intellij-common/blob/06f9ee4999320f9c542b8558ce2e1e53aab3340e/src/main/java/com/redhat/devtools/intellij/common/compat/PopupHandlerAdapter.java#L29

Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/WindowToolFactory.java]: done.

**Review by adietish — APPROVED** [review]
lgtm

**Comment by sbouchet** [comment]
@adietish : any chances to release intellij-common for the release? i'm not fan to release with SNAPSHOT deps.

**Comment by sbouchet** [comment]
> @adietish : any chances to release intellij-common for the release? i'm not fan to release with SNAPSHOT deps.

resolved with #534

**Pull request merged**